### ISSUE TITLE
[bug] parens and staticlookup

### ISSUE THREAD

**Issue by evilebottnawi** [glayzzle/php-parser#185]
```php
$var = ($var[0])::foo;
$var = ($var[0][1])::foo;
$var = ($var[0])[1]::foo;
$var = (($var[0])[1])::foo;
$var = (new Foo())::bar;
```

Problem exists in master

**Comment by evilebottnawi** [comment]
Here other big bug 
```php
get_class($this->resource)::$wrap;
```

**Comment by ichiriac** [comment]
not sure yet, but seems that this bug existed before

**Comment by evilebottnawi** [comment]
maybe :disappointed:

**Comment by evilebottnawi** [comment]
Need also fix:
```php
get_class($var)::$$$$$property;
```

**Comment by ichiriac** [comment]
Ok, for $$$$$property I'm not enough permissive here : https://github.com/glayzzle/php-parser/blob/master/src/parser/variable.js#L154

**Comment by ichiriac** [comment]
other bugs are from : https://github.com/glayzzle/php-parser/blob/master/src/parser/variable.js#L232 and introduced here https://github.com/glayzzle/php-parser/issues/179 

I've forced the call as the bugfix request was over staticlookup with call functions, but here we are working on another branch :

```
simple_variable:
		T_VARIABLE			{ $$ = $1; }
	|	'$' '{' expr '}'	{ $$ = $3; }
	|	'$' simple_variable	{ $$ = zend_ast_create(ZEND_AST_VAR, $2); }
;
static_member:
		class_name T_PAAMAYIM_NEKUDOTAYIM simple_variable
			{ $$ = zend_ast_create(ZEND_AST_STATIC_PROP, $1, $3); }
	|	variable_class_name T_PAAMAYIM_NEKUDOTAYIM simple_variable
			{ $$ = zend_ast_create(ZEND_AST_STATIC_PROP, $1, $3); }
;
```

In principle it's fixed, but can you please provide more tests and check if it's not too permissive.

**Comment by ichiriac** [comment]
released in `3.0.0-prerelease.8`
